kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -239,14 +243,21 @@ public Optional<SetTransaction> getSetTxnOpt() {
         return setTxnOpt;
     }
 
+    /**
+     * Generates a timestamp which is greater than the commit timestamp of the readSnapshot.
```

Comment:
We should also mention that this can result in an additional file read and that this will only happen if ICT is enabled.